src/app/GitUI/LeftPanel/FavoriteBranchesCache.cs

```diff
@@ -0,0 +1,257 @@
+        {
+            return;
+        }
+
+        lock (_lock)
+        {
+            if (_favorites.Add(branch))
+            {
+                Save();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Checks if a branch is marked as a favorite.
+    /// </summary>
+    /// <param name="objectId">The unique ObjectId of the branch's latest commit.</param>
+    /// <param name="branchName">The name of the branch (fully qualified).</param>
+    /// <returns><c>true</c> if the branch is a favorite; otherwise, <c>false</c>.</returns>
+    internal bool Contains(ObjectId objectId, string branchName)
+    {
+        BranchIdentifier branch = new(objectId, branchName);
+
+        if (!BranchIdentifier.IsValid(branch))
+        {
+            return false;
+        }
+
+        LoadIfNeeded();
+
+        lock (_lock)
+        {
+            return _favorites.Contains(branch);
+        }
+    }
+
+    /// <summary>
+    /// Cleans up the favorites list by removing branches that no longer exist.
+    /// </summary>
+    /// <param name="branches">A list of branches to retain in the favorites list.</param>
+    internal void CleanUp(IReadOnlyList<IGitRef> branches)
+    {
+        List<BranchIdentifier> removableList = new();
+
+        lock (_lock)
+        {
+            foreach (IGitRef? branch in branches)
+            {
+                if (branch.ObjectId != null)
+                {
+                    BranchIdentifier item = new(branch.ObjectId, branch.Name);
+
+                    if (!_favorites.Contains(item))
+                    {
+                        removableList.Add(item);
+                    }
+                }
+            }
+
+            removableList.ForEach(item => _favorites.Remove(item));
+            Save();
+        }
+    }
+
+    /// <summary>
+    /// Loads the list of favorite branches from the configuration file into memory.
+    /// </summary>
+    /// <remarks>
+    /// This method reads the configuration file in JSON format and deserializes it into a list of `BranchIdentifier`
+    /// objects. If the file does not exist, the method exits without taking any action.
+    /// </remarks>
+    internal void Load()
+    {
+        if (!File.Exists(ConfigFile))
+        {
+            return;
+        }
+
+        try
+        {
+            string json;
+
+            lock (_lock)
+            {
+                json = _fileSystem.File.ReadAllText(ConfigFile);
+            }
+
+            BranchIdentifier[]? deserialized = JsonSerializer.Deserialize<BranchIdentifier[]>(json, _jsonOptions);
+
+            if (deserialized is not null)
+            {
+                lock (_lock)
+                {
+                    _favorites.Clear();
+                    _favorites.UnionWith(deserialized);
+                }
+            }
+
+            _isLoaded = true;
+        }
+        catch (Exception ex)
+        {
+            Debug.WriteLine($"Failed to load favorites: {ex.Message}");
```

Comment:
```suggestion
            Trace.WriteLine($"Failed to load favorites: {ex.Message}");
```

Ditto other places, so the message can be seen in a DebugView (or equivalent).